bindings/matrix-sdk-ffi/src/widget.rs

```diff
@@ -303,10 +292,26 @@ pub fn get_element_call_required_permissions(
             WidgetEventFilter::StateWithType { event_type: StateEventType::RoomCreate.to_string() },
         ],
         send: vec![
-            // To send the call participation state event (main MatrixRTC event)
+            // To send the call participation state event (main MatrixRTC event).
+            // This is required for legacy state events (using only one event for all devices with
+            // a membership array). TODO: remove once legacy call member events are
+            // sunset.
+            WidgetEventFilter::StateWithTypeAndStateKey {
+                event_type: StateEventType::CallMember.to_string(),
+                state_key: own_user_id.clone(),
+            },
+            // `delayed_event`` version for session memberhips
+            // [MSC3779](https://github.com/matrix-org/matrix-spec-proposals/pull/3779), with no leading underscore.
             WidgetEventFilter::StateWithTypeAndStateKey {
                 event_type: StateEventType::CallMember.to_string(),
-                state_key: own_user_id,
+                state_key: format!("{}_{}", own_user_id, own_device_id),
```

Comment:
Could be:

```suggestion
                state_key: format!("{own_user_id}_{own_device_id}"),
```

Probably more Rust idiomatic.